### PR TITLE
644 run fluka simulation

### LINKED ISSUE #644: Add feature to run Fluka simulation from backend API call
This task should include:
 -  [ ] adding missing parameters to the API
 -  [ ] improving openapi.yml swagger documentation
 -  [ ] adding Fluka fake binary to the codebase
 -  [ ] adding runner method to execute Fluka binary when backend is requested to do so
 -  [ ] return hardcoded JSON output
 -  [ ] add proper pytest stuff

With such changes we can make first tests with UI

### PR DESCRIPTION
**Pull request by hendzeld**
Resolves #644

### CODE REVIEW

File: yaptide/celery/tasks.py [hunk 1]
```diff
@@ -56,8 +56,9 @@ def run_single_simulation(self,
                           task_id: str,
                           update_key: str = None,
                           simulation_id: int = None,
-                          keep_tmp_files: bool = False) -> dict:
-    """Function running single shieldhit simulation"""
+                          keep_tmp_files: bool = False,
+                          sim_type: str = 'shieldhit') -> dict:
+    """Function running single simulation"""
```
Comment (hendzeld) [inline yaptide/celery/tasks.py:61]: I am not sure if we want to add parameters here.
Maye, each new simulator should have its own task?

Comment (grzanka) [reply yaptide/celery/tasks.py:61]: If the code would be very different then it may be a good idea

File: yaptide/celery/utils/pymc.py [hunk 2]
```diff
@@ -61,6 +64,70 @@ def run_shieldhit(dir_path: Path, task_id: str) -> dict:
...
+
+    class UpdateFlukaRandomSeed(Protocol):
+        """Protocol for updating random seed in fluka input file"""
+
+        def __call__(self, file_path: str, rng_seed: int) -> None:
+            """Updates random seed in fluka input file"""
+
+    random_seed = int(task_id.split("_")[-1])
+    # propably should be protected instead of private
+    update_fulka_function: UpdateFlukaRandomSeed = Runner._Runner__update_fluka_input_file  # pylint: disable=W0212
```
Comment (grzanka) [inline yaptide/celery/utils/pymc.py]: fulka -> fluka

**Comment by grzanka** [comment]
@hendzeld there is a merge conflict not

File: yaptide/celery/tasks.py (hunk 1 above)
Comment (hendzeld) [reply yaptide/celery/tasks.py:61]: It seems that for now it is very similar.

I was concerned more about protocol(task) compatibility

File: tests/integration/test_celery.py [hunk 3]
```diff
@@ -71,3 +72,40 @@ def test_celery_run_simulation(celery_app,
...
+        ) for i in range(payload_dict["ntasks"])
+    ])
+
+    workflow = chord(map_group, merge_results.s())
+    job = workflow.delay()
+    logging.info("Waiting for run_simulation task to finish")
+    result: dict = job.wait()
+    logging.info("run_simulation task finished")
+    assert 'logfiles' not in result.keys()
+    assert 'estimators' in result.keys()
```
Comment (grzanka) [inline tests/integration/test_celery.py]: is there some output in the estimators ? if yes, then lets check at least if the output is not empty

File: run_worker.sh [hunk 4]
```diff
@@ -22,4 +22,11 @@ if [ -z "$S3_TOPAS_BUCKET" ] || [ -z "$S3_TOPAS_KEY" ] || [ -z "$S3_TOPAS_VERSIO
...
 
-celery --app yaptide.celery.worker worker -E --loglevel="$LOG_LEVEL_ROOT" -P eventlet --hostname yaptide-worker
+# Copy fluka fake simulator from yaptide dir
+echo "Copying fluka fake simulator from yaptide dir"
+
+mkdir -p /simulators/fluka/bin
+cp ./yaptide/fake/rfluka /simulators/fluka/bin/rfluka
+chmod +x /simulators/fluka/bin/rfluka
+
+celery --app yaptide.celery.worker worker -E --loglevel=info -P eventlet --hostname yaptide-worker
```
Comment (grzanka) [inline run_worker.sh]:
```suggestion
celery --app yaptide.celery.worker worker -E --loglevel="$LOG_LEVEL_ROOT" -P eventlet --hostname yaptide-worker
```

**Review by grzanka — APPROVED** [review]

**Pull request merged**